common/unsafe/src/main/java/org/apache/spark/unsafe/memory/HeapMemoryAllocator.java

```diff
@@ -46,9 +47,10 @@ private boolean shouldPool(long size) {
 
   @Override
   public MemoryBlock allocate(long size) throws OutOfMemoryError {
-    if (shouldPool(size)) {
```

Comment:
I think we don't need to create a new method:
```
int numWords = (int) ((size + 7) / 8); // add some overflow check.
int alignedSize = numWords * 8;
if (shouldPool(alignedSize)) {
  ...
}
long[] array = new long[numWords];
```

Reply:
I aggree with you .
I hava updated,thanks